DS13/code/modules/EMH/EMH.dm

```diff
@@ -0,0 +1,326 @@
+	user.forceMove(get_turf(A))
+	playsound(user.loc, 'DS13/sound/effects/holofizzle.ogg', 100, 0)
+	var/obj/machinery/emh_emitter/S = A
+	emh = null
+	S.emh = user //transfer silently.
+	START_PROCESSING(SSfastprocess, S)
+
+
+/obj/machinery/emh_emitter/process()
+	if(!emh || !is_occupied()) //If we don't have an EMH, no need to process
+		STOP_PROCESSING(SSfastprocess, src)
+		return
+	if(QDELETED(emh)) //Same goes here
+		emh = null
+		icon_state = "emh-off"
+		return
+	icon_state = "emh-on" //Well we have an EMH, let's check if he's nearby
+	if(get_dist(emh, src) >= range || emh.z != z) //Out of range, try find the EMH another emitter or tele him back to us.
+		var/obj/machinery/emh_emitter/closest = null //Find our closest distance
+		var/closest_dist = 10 //Distance of closest emitter
+		for(var/obj/machinery/emh_emitter/em in GLOB.machines)
+			if(em == src || !emh in orange(em, em.range))
+				return
+			if(get_dist(emh, em) <= closest_dist && em.z == emh.z)
+				closest = em
+				closest_dist = get_dist(emh, em)
+		if(closest)
+			if(get_dist(emh, closest) >= closest.range || emh.z != closest.z) //if he's moved since the last process cycle.
+				to_chat(emh, "You cannot move beyond the range of your holo-emitters!")
+				emh.forceMove(get_turf(closest))
+			closest.emh = emh //Transfer his program seamlessly
+			emh = null
+			closest.icon_state = "emh-on"
+			icon_state = "emh-off"
+			START_PROCESSING(SSfastprocess, closest)
+		else
+			to_chat(emh, "You cannot move beyond the range of your holo-emitters!")
+			emh.forceMove(get_turf(src))
+
+/obj/machinery/emh_emitter/proc/activate(var/mob/living/carbon/human/species/holographic/transferred = null) //Activate, if we're not transferring an existing EMH over, then make a new one!
+	if(transferred)
+		emh = transferred
+		new /obj/effect/temp_visual/dir_setting/ninja/cloak(get_turf(emh), emh.dir)
+		transferred.forceMove(get_turf(src))
+		to_chat(transferred, "You have been transferred to [get_area(src)]")
+		emh.say("Please state the nature of the medical emergency.")
+		playsound(loc, 'DS13/sound/effects/medicalemergency.ogg', 50, 0)
+		START_PROCESSING(SSfastprocess,src)
+		return
+	if(is_occupied())
+		say("Unable to comply, there is already an EMH active on the network.")
+		return
+	if(emh) //AFK emh can get replaced with a live one!
+		qdel(emh)
+	emh = new /mob/living/carbon/human/species/holographic(get_turf(src)) //Please state th-
+	emh.name = "Emergency Medical Hologram"
+	emh.real_name = "Emergency Medical Hologram"
+	emh.alpha = 0 //So that it "pops" out of nowhere with a mind in it, or else it deletes
+	emh.equipOutfit(/datum/outfit/emh)
+	offer_control_emh(emh)
+	if(emh.client && emh.mind) //Hitchiking off of this until I can make a new jobban
+		emh.say("Please state the nature of the medical emergency.")
+		emh.hair_style = random_hair_style(emh.gender)
+		playsound(loc, 'DS13/sound/effects/medicalemergency.ogg', 50, 0)
+		emh.alpha = 255
```

Comment:
this should be in `/mob/living/carbon/human/species/holographic` init or postspawn or however your species code works